### PR TITLE
GH-224: Avoid hack for populating go to menu entries for Xpect tests 

### PR DESCRIPTION
**Pull request by qtran-n4**
Task: https://github.com/eclipse/Xpect/issues/224

### CODE REVIEW

**Comment by qtran-n4** [comment]
@meysholdt 

- The labels for Xpect methods are displayed in the tree ResultView. 
- The context menu `Go to Xpect method` is added 

However, the following does not work:
- Double click
- `Go to File`
because these are implementations of JDT JUnit.

screen shot 2018-01-29 at 15 35 45

File: org.eclipse.xtext.example.domainmodel.xpect.tests/src/org/eclipse/xtext/example/domainmodel/xpect/tests/scoping/DMScopingTest.java [hunk 1]
```diff
@@ -21,3 +21,4 @@
 public class DMScopingTest extends ScopingTest {
 
 }
+
```
Comment (yoosiba) [inline org.eclipse.xtext.example.domainmodel.xpect.tests/src/org/eclipse/xtext/example/domainmodel/xpect/tests/scoping/DMScopingTest.java]: add new line

File: org.eclipse.xpect.ui.junit/src/org/eclipse/xpect/ui/junit/GotoXpectMethodAction.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		if (selection instanceof TreeSelection) {
+			TreeSelection t = (TreeSelection) selection;
+			TestElement testElement = (TestElement) t.getFirstElement();
+			if (testElement instanceof TestCaseElement) {
+				testCaseElement = (TestCaseElement) testElement;
+			}
+		}
+		action.setEnabled(testCaseElement != null);
+	}
+}
```
Comment (yoosiba) [inline org.eclipse.xpect.ui.junit/src/org/eclipse/xpect/ui/junit/GotoXpectMethodAction.java]: new line

**Comment by qtran-n4** [comment]
@meysholdt Good catch Moritz. 
I am stuck then :( How can we calculate the URI to an Xpect method while providing short labels for Xpect methods in the ResultView tree. Any idea?

**Comment by meysholdt** [comment]
To run an individual test maybe Xpect could filter out all sibling tests but *keep* the parent test suites. However, if we're unlucky Xpect does this already and it's JDT that removes the parent test suites.

**Comment by qtran-n4** [comment]
It's JDT that does not create a TestSuiteElement node if a single Xpect test is executed (cf. `TestRunSession#createTestElement`)

```java
	public TestElement createTestElement(TestSuiteElement parent, String id, String testName, boolean isSuite, int testCount) {
		TestElement testElement;
		if (isSuite) {
			TestSuiteElement testSuiteElement= new TestSuiteElement(parent, id, testName, testCount);
			testElement= testSuiteElement;
			if (testCount > 0)
				fIncompleteTestSuites.add(new IncompleteTestSuite(testSuiteElement, testCount));
		} else {
			testElement= new TestCaseElement(parent, id, testName);
		}
		fIdToTest.put(id, testElement);
		return testElement;
	}
```

**Comment by qtran-n4** [comment]
@meysholdt I have included in resource URI in the labels for Xpect test methods. For readability, I show the Xpect test method name + optional comment on the right hand side and put the resource URI inside a special brackets. 

screen shot 2018-02-06 at 16 40 18

**Comment by qtran-n4** [comment]
@yoosiba @meysholdt Please have a second look

**Pull request merged**